op-node/rollup/derive/span_batch.go

```diff
@@ -582,15 +618,38 @@ func (b *SpanBatch) GetSingularBatches(l1Origins []eth.L1BlockRef, l2SafeHead et
 }
 
 // NewSpanBatch converts given singularBatches into SpanBatchElements, and creates a new SpanBatch.
-func NewSpanBatch(singularBatches []*SingularBatch) *SpanBatch {
-	spanBatch := &SpanBatch{}
+func NewSpanBatch(genesisTimestamp uint64, chainID *big.Int) *SpanBatch {
+	return &SpanBatch{
+		GenesisTimestamp: genesisTimestamp,
+		ChainID:          chainID,
+
+		// set the spanBatchTxs structure with defaults
+		sbtxs: &spanBatchTxs{
+			contractCreationBits: big.NewInt(0),
+			yParityBits:          big.NewInt(0),
+			txSigs:               []spanBatchSignature{},
+			txNonces:             []uint64{},
+			txGases:              []uint64{},
+			txTos:                []common.Address{},
+			txDatas:              []hexutil.Bytes{},
+			txTypes:              []int{},
+			protectedBits:        big.NewInt(0),
+		},
```

Comment:
In a follow-up, we should consider using a `sync.Pool` of `*spanBatchTxs`. There's a lot of allocations and slice growth that would quickly amortize over the runtime of a batcher with a `sync.Pool`. We'd just need to give it a reset method.

This method is already used e.g. in `batch.go` for the encode buffers. We could maybe also use the same trick for some other buffers like compression buffers.

Reply:
And note that there is `newSpanBatchTxs`